transport/request.hh

```diff
@@ -71,130 +73,237 @@ public:
+            }
+            names.emplace_back(std::move(name).value());
+            utils::result_with_eptr<value_view_and_unset> vv = read_value_view(version);
+            if (!vv) [[unlikely]] {
+                return bo::failure(std::move(vv).error());
+            }
+            auto&& [value, is_unset] = std::move(vv).value();
             values.emplace_back(std::move(value));
             unset.emplace_back(is_unset);
         }
+        return bo::success();
     }
 
-    void read_string_list(std::vector<sstring>& strings) {
-        uint16_t size = read_short();
-        strings.reserve(size);
-        for (uint16_t i = 0; i < size; i++) {
-            strings.emplace_back(read_string());
+    utils::result_with_eptr<void> read_string_list(std::vector<sstring>& strings) {
+        utils::result_with_eptr<uint16_t> size = read_short();
+        if (!size) [[unlikely]] {
+            return bo::failure(std::move(size).error());
+        }
+        strings.reserve(size.value());
+        for (uint16_t i = 0; i < size.value(); i++) {
+            utils::result_with_eptr<sstring> str = read_string();
+            if (!str) [[unlikely]] {
+                return bo::failure(std::move(str).error());
+            }
+            strings.emplace_back(std::move(str).value());
         }
+        return bo::success();
     }
 
-    void read_value_view_list(uint8_t version, std::vector<cql3::raw_value_view>& values, cql3::unset_bind_variable_vector& unset) {
-        uint16_t size = read_short();
-        values.reserve(size);
-        unset.reserve(size);
-        for (uint16_t i = 0; i < size; i++) {
-            auto&& [value, is_unset] = read_value_view(version);
+    utils::result_with_eptr<void> read_value_view_list(uint8_t version, std::vector<cql3::raw_value_view>& values, cql3::unset_bind_variable_vector& unset) {
+        utils::result_with_eptr<uint16_t> size = read_short();
+        if (!size) [[unlikely]] {
+            return bo::failure(std::move(size).error());
+        }
+        values.reserve(size.value());
+        unset.reserve(size.value());
+        for (uint16_t i = 0; i < size.value(); i++) {
+            utils::result_with_eptr<value_view_and_unset> vv = read_value_view(version);
+            if (!vv) [[unlikely]] {
+                return bo::failure(std::move(vv).error());
+            }
+            auto&& [value, is_unset] = std::move(vv).value();
             values.emplace_back(std::move(value));
             unset.emplace_back(is_unset);
         }
+        return bo::success();
     }
 
-    db::consistency_level read_consistency() {
-        return wire_to_consistency(read_short());
+    utils::result_with_eptr<db::consistency_level> read_consistency() {
+        utils::result_with_eptr<uint16_t> v = read_short();
+        if (!v) [[unlikely]] {
+            return bo::failure(std::move(v).error());
+        }
+        return wire_to_consistency(v.value());
```

Comment:
I think the patch goes a little bit too far. There are expected exceptions like bad protocol numbers (since they are part of the negotiation). There are unexpected exceptions (like failing to read a short or something) since drivers usually behave.

There will be some extra instructions involved in all this checking, so some performance impact.